testsuite/forge/src/backend/k8s/swarm.rs

```diff
@@ -0,0 +1,289 @@
+            .get_mut(&id)
+            .map(|v| v as &mut dyn Validator)
+    }
+
+    fn full_nodes<'a>(&'a mut self) -> Box<dyn Iterator<Item = &'a dyn FullNode> + 'a> {
+        Box::new(self.fullnodes.values().map(|v| v as &'a dyn FullNode))
+    }
+
+    fn full_nodes_mut<'a>(&'a mut self) -> Box<dyn Iterator<Item = &'a mut dyn FullNode> + 'a> {
+        Box::new(
+            self.fullnodes
+                .values_mut()
+                .map(|v| v as &'a mut dyn FullNode),
+        )
+    }
+
+    fn full_node(&self, id: PeerId) -> Option<&dyn FullNode> {
+        self.fullnodes.get(&id).map(|v| v as &dyn FullNode)
+    }
+
+    fn full_node_mut(&mut self, id: PeerId) -> Option<&mut dyn FullNode> {
+        self.fullnodes.get_mut(&id).map(|v| v as &mut dyn FullNode)
+    }
+
+    fn add_validator(&mut self, _id: PeerId) -> Result<PeerId> {
+        todo!()
+    }
+
+    fn remove_validator(&mut self, _id: PeerId) -> Result<()> {
+        todo!()
+    }
+
+    fn add_full_node(&mut self, _id: PeerId) -> Result<()> {
+        todo!()
+    }
+
+    fn remove_full_node(&mut self, _id: PeerId) -> Result<()> {
+        todo!()
+    }
+
+    fn chain_info(&mut self) -> ChainInfo<'_> {
+        let url = self.get_url();
+        ChainInfo::new(
+            &mut self.root_account,
+            &mut self.treasury_compliance_account,
+            &mut self.designated_dealer_account,
+            url,
+            self.chain_id,
+        )
+    }
+}
+
+fn k8s_retry_strategy() -> impl Iterator<Item = Duration> {
+    diem_retrier::exp_retry_strategy(1000, 5000, 30)
+}
+
+#[derive(Clone, Debug)]
+pub struct KubeService {
+    pub name: String,
+    pub host_ip: String,
+}
+
+impl TryFrom<Service> for KubeService {
+    type Error = anyhow::Error;
+
+    fn try_from(service: Service) -> Result<Self, Self::Error> {
+        let metadata = service.metadata;
+        let name = metadata
+            .name
+            .ok_or_else(|| format_err!("node name not found"))?;
+        let spec = service
+            .spec
+            .ok_or_else(|| format_err!("spec not found for node"))?;
+        let host_ip = spec.cluster_ip.unwrap_or_default();
+        Ok(Self { name, host_ip })
+    }
+}
+
+pub async fn list_services(client: K8sClient) -> Result<Vec<KubeService>> {
+    let node_api: Api<Service> = Api::all(client);
+    let lp = ListParams::default();
+    let services = node_api.list(&lp).await?.items;
+    services.into_iter().map(KubeService::try_from).collect()
+}
+
+fn parse_node_id(s: &str) -> Result<usize> {
+    let v = s.split('-').collect::<Vec<&str>>();
+    if v.len() < 5 {
+        return Err(format_err!("Failed to parse {:?} node id format", s));
+    }
+    let idx: usize = v[0][3..].parse().unwrap();
+    Ok(idx)
+}
+
+// TODO remove hard code key
+fn load_root_key() -> Ed25519PrivateKey {
+    let composite_key = base64::decode(
+        "iuRk67ESWWbbXqJu33GK0VGzGO3IOYYDT73lWE3WwZhuLNdkknw80heocg/Ktg4bie1ssmHx1GaJeD5j8LqBAA==",
```

Comment:
You had to query this key from a vault service at some point, correct? Couldn't we just have this service do that querying instead of hardcoding it?